homeassistant/components/medcom_ble/config_flow.py

```diff
@@ -0,0 +1,153 @@
+
+    async def async_step_bluetooth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm discovery."""
+        # We always will have self._discovery_info be a BluetoothServiceInfo at this point
+        # and this helps mypy not complain
+        assert self._discovery_info is not None
+
+        if user_input is None:
+            name = self._discovery_info.name or self._discovery_info.address
+            return self.async_show_form(
+                step_id="bluetooth_confirm",
+                description_placeholders={"name": name},
+            )
+
+        return await self.async_step_check_connection()
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user step to pick discovered device."""
+        if user_input is not None:
+            address = user_input[CONF_ADDRESS]
+            await self.async_set_unique_id(address, raise_on_progress=False)
+            self._abort_if_unique_id_configured()
+            self._discovery_info = self._discovered_devices[address]
+            self.context["title_placeholders"] = {
+                "name": self._discovery_info.name,
+            }
+            return await self.async_step_check_connection()
+
+        current_addresses = self._async_current_ids()
+        for discovery_info in async_discovered_service_info(self.hass):
+            address = discovery_info.address
+            if address in current_addresses or address in self._discovered_devices:
+                _LOGGER.debug(
+                    "Detected a device that's already configured: %s", address
+                )
+                continue
+
+            if INSPECTOR_SERVICE_UUID not in discovery_info.service_uuids:
+                continue
+
+            self._discovered_devices[discovery_info.address] = discovery_info
+
+        if not self._discovered_devices:
+            return self.async_abort(reason="no_devices_found")
+
+        titles = {
+            address: discovery.name
+            for address, discovery in self._discovered_devices.items()
+        }
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_ADDRESS): vol.In(titles),
+                },
+            ),
+        )
+
+    async def async_step_check_connection(self) -> FlowResult:
+        """Check we can connect to the device before considering the configuration is successful."""
+        # We always will have self._discovery_info be a BluetoothServiceInfo at this point
+        # and this helps mypy not complain
+        assert self._discovery_info is not None
+
+        _LOGGER.debug("Checking device connection: %s", self._discovery_info.name)
+        try:
+            await self._get_device_data(self._discovery_info)
+        except (BleakError, AbortFlow):
```

Comment:
```suggestion
        except BleakError:
```

We want AbortFlow to raise, no need to trap it